src/core/server/csp/config.ts

```diff
@@ -7,28 +7,145 @@
+    if (!allowNonce && value.startsWith('nonce-')) {
+      return `using "nonce-*" is considered insecure and is not allowed`;
+    }
+    if (!allowNone && (value === `none` || value === `'none'`)) {
+      return `using "none" would conflict with Kibana's default csp configuration and is not allowed`;
+    }
+  };
+};
+
+const configSchema = schema.object(
+  {
+    rules: schema.maybe(schema.arrayOf(schema.string())),
+    script_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    worker_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    style_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    connect_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    default_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    font_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    frame_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    img_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    frame_ancestors: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    report_uri: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    report_to: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    strict: schema.boolean({ defaultValue: true }),
+    warnLegacyBrowsers: schema.boolean({ defaultValue: true }),
+    disableEmbedding: schema.oneOf([schema.literal<boolean>(false)], { defaultValue: false }),
+  },
+  {
+    validate: (cspConfig) => {
+      if (cspConfig.rules && hasDirectiveSpecified(cspConfig)) {
+        return `"csp.rules" cannot be used when specifying per-directive additions such as "script_src", "worker_src" or "style_src"`;
+      }
+      if (
+        cspConfig.strict &&
+        (cspConfig.script_src.includes(`unsafe-inline`) ||
+          cspConfig.script_src.includes(`'unsafe-inline'`))
+      ) {
+        return 'cannot use `unsafe-inline` for `script_src` when `csp.strict` is true';
+      }
```

Comment:
I just realized that we have a similar problem if `csp.script_src: [unsafe-inline]` is used along with `csp.warnLegacyBrowsers: true` (default):

every page load in a modern browser results in this nasty warning toast

![image](https://user-images.githubusercontent.com/5295965/123126436-6217df00-d417-11eb-8751-6f34ca4c4194.png)

So perhaps we should also add a check for that and don't allow `unsafe-inline` to be used in conjunction with `warnLegacyBrowsers` either.